src/Chapter3.hs

```diff
@@ -970,6 +1111,44 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
+data WeekDay
+	= Monday
+	| Tuesday
+	| Wednesday
+	| Thursday
+	| Friday
+	| Saturday
+	| Sunday deriving (Show, Read, Eq, Ord, Enum)
+
+isWeekend :: WeekDay -> Bool
+isWeekend day = case day of
+	Saturday -> True
+	Sunday -> True
+	_ -> False
+
+nextDay :: WeekDay -> WeekDay
+nextDay day = case day of
+	Sunday -> Monday
```

Comment:
That is right!
You could also use the instance of `Bounded` in here:

```suggestion
    | day == maxBound = minBound
```

This is also a bit more maintainable. If you decide, for example, to have the "American" system of week days in the data type constructors, you won't need to change this function at all! 🙂